src/network.rs

```diff
@@ -1,3 +1,39 @@
 //! Network specific utils
 
+use std::fmt::Debug;
+
+use alloy_primitives::U256;
+use alloy_rpc_types::{Header, Transaction};
+use revm::primitives::{BlockEnv, SpecId, TxEnv};
+
+use crate::mv_memory::MvMemory;
+
+/// A chain ID (u64) associated with relevant utils.
+pub trait PevmChain {
```

Comment:
This should be `PevmNetwork` as we are in `src/network`. If we go with `PevmChain` then this should be renamed to `src/chain`. I have no clear preference at the moment.